### PR TITLE
Zoom Client

### PR DESCRIPTION
**Pull request by frankmeszaros**
Essential Zoom Client Support
- ability to get list of users
- ability to get user details
- ability to get list of meetings 
- ability to get meeting details
- ability to get list of operation logs
- ability to get cloud recordings
- ability to get cloud recording details for a given meeting

### CODE REVIEW

File: vectrix/clients/zoom.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+    def __get_pagination_path(self,
+                              base_path,
+                              next_page_token=None,
+                              page_number=None,
+                              page_size=30):
+
+        pagination_params = f"page_size={page_size}"
+        if page_number:
+            pagination_params = f"{pagination_params}&page_number={page_number}"
```
Comment (frankmeszaros) [inline vectrix/clients/zoom.py:48]: this value is sometimes on paginated requests but sometimes not, making it optional

File: vectrix/clients/zoom.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        page_size = 30
+        path = self.__get_pagination_path(base_path,
+                                          page_number=page_number,
+                                          page_size=page_size)
+
+        paginated_response = self.__request(verb, path)
+        next_page_token = paginated_response.get("next_page_token")
+        items = paginated_response.get(key)
+
+        while next_page_token != '':
```
Comment (frankmeszaros) [inline vectrix/clients/zoom.py:73]: not sure what level of control we want end-client to have so i defaulted to a fetch-all option. open to other suggestions though as I am still somewhat undecided on the optimal approach. Didn't seem like we want to make pagination something the user has to deal with.

File: vectrix/clients/zoom.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+
+class ZoomClient:
+    """
+    A thin client around Zoom API http requests
+    """
+    def __init__(self, access_token):
+        self.access_token = access_token
+
+    def __request(self, verb, path):
```
Comment (Lewisecurity) [inline vectrix/clients/zoom.py:27]: In an effort for extensibility in detection packs, can we expose methods like `request` and `paginated_request` or just 1 big `request(paginated=True)` to the detection packs so when there are other API endpoints being hit that the client doesn't have native support for then we can still call on them?

Perhaps something like: `def request(self, path, paginated=False):`

Then, if a detection pack author does `paginated=True` then it'll keep iterating over.

File: vectrix/clients/zoom.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        conn = http.client.HTTPSConnection("api.zoom.us")
+        headers = {
+            'authorization': f"Bearer {self.access_token}",
+            'content-type': "application/json"
+        }
+
+        conn.request(verb, path, headers=headers)
+        res = conn.getresponse()
```
Comment (Lewisecurity) [inline vectrix/clients/zoom.py:35]: Perhaps out of scope for this PR, but we will want to have solid support for request failure (ie. 500), request timeout (network issue), rate limiting, and authorization issues before passing back the response.

(I'm totally happy to take a swing at this myself, def could be out of scope for this PR!)

File: vectrix/clients/zoom.py (hunk 2 above)
Comment (Lewisecurity) [reply vectrix/clients/zoom.py:73]: I think the current approach is perfect!

File: vectrix/clients/zoom.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    def get_meeting_recordings(self, meeting_id):
+        """
+         Get all the recordings from a meeting or webinar instance
+        """
+        return self.__request("GET", f"v2/meetings/{meeting_id}/recordings")
+
+    def get_im_groups(self):
+        return self.__request("GET", "/v2/im/groups")
+
+    def get_operation_logs(
```
Comment (Lewisecurity) [inline vectrix/clients/zoom.py:170]: Awesome method here.

**Review by Lewisecurity — COMMENTED** [review]
Great start! Besides my comments we just need to get some tests written 😄

File: vectrix/clients/zoom.py [hunk 6]
```diff
@@ -0,0 +1,181 @@
+import http.client
```
[the remark this answers is not in the document]
Comment (frankmeszaros) [reply vectrix/clients/zoom.py:1]: for sure! This was selected purely based on the zoom example docs. I can refactor this accordingly. thanks

File: setup.py [hunk 7]
```diff
@@ -23,41 +23,36 @@ def run(self):
...
-        'Programming Language :: Python :: 3.6',
-        'Programming Language :: Python :: 3.7',
-        'Programming Language :: Python :: 3.8',
-    ],
-    python_requires='>=3',
-    cmdclass={
-        'verify': VerifyVersionCommand,
-    }
-)
+setup(name='vectrix',
```
Comment (frankmeszaros) [inline setup.py:30]: formatter changes. @Lewisecurity LMK your preferred set up and will be happy to conform. 

Picked `yapf` and `flake8` randomly and have no attachment to either.

File: tox.ini [hunk 8]
```diff
@@ -7,5 +7,6 @@ deps =
     pytest
     pytest-cov
     pytest-mock
+    requests-mock
```
Comment (frankmeszaros) [inline tox.ini:10]: when you need to add new dev dependencies, be sure to add them to tox.

**Review by Lewisecurity — APPROVED** [review]
Can you just update the version in `setup.py` and then we should be good to go!

**Review by Lewisecurity — APPROVED** [review]

**Pull request merged**